File: spring/core/SpringBeanLifecycles/src/main/java/demos/lifecycle/prototype/scope/PricingEngineMock.java
package demos.lifecycle.prototype.scope;

import demos.lifecycle.common.PricingEngine;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;

@Component("pricing")
@Scope("prototype")
public class PricingEngineMock implements PricingEngine {
    public double price(String itemNo, int quantity) {
        return quantity * 5.50;
    }
}

File: spring/core/AnnotationDrivenDIWithKotlin/src/main/kotlin/demos/spring/notes/v9/PricingEngineMock.kt
package demos.spring.notes.v9

import demos.spring.notes.common.PricingEngine
import org.springframework.context.annotation.Scope
import org.springframework.stereotype.Component

@Component("pricing")
@Scope("prototype")
class PricingEngineMock : PricingEngine {
    override fun price(itemNo: String, quantity: Int): Double {
        return quantity * 5.50
    }
}
